apps/server/src/modules/tool/external-tool/entity/external-tool.entity.ts

```diff
@@ -42,6 +42,7 @@ export interface ExternalToolEntityProps {
 }
 
 @Entity({ tableName: 'external-tools' })
+@Index({ properties: ['config_clientId'], options: { unique: true, sparse: true } })
```

Comment:
Wenn es nur einen key betrifft und kein combined ist legen wir den Index direkt dort ab.

@Index()
@Unique({options: { sparse: true }})
config_clientId: ...

Allerdings fehlt der key config_clientId in der Entität und muss dann auch hinzugefügt werden.

Reply:
@CeEv thats not possible due to the nature of how this is built, it can be multiple types of config, some have that, some don't, in the past we were forced to add that index in each possible type, that caused more problems later downstream and in that instance that field wasn't even nullable

Reply:
If i add this Index and Unique with sparse to the Oauth2ToolConfigEntity and the migration job for index faild